src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_item.c

```diff
@@ -1,85 +1,1000 @@
 /*
  * File: z_kaleido_item.c
  * Overlay: ovl_kaleido_scope
- * Description:
+ * Description: Pause Menu - Item Page
  */
 
 #include "z_kaleido_scope.h"
+#include "interface/parameter_static/parameter_static.h"
 
-extern UNK_TYPE D_02001360;
-extern UNK_TYPE D_020044A0;
-extern UNK_TYPE D_02004AA0;
-extern UNK_TYPE D_0200B998;
-extern UNK_TYPE D_08062000;
-extern UNK_TYPE D_08064340;
-extern UNK_TYPE D_0B000000;
-extern UNK_TYPE D_0C000000;
-extern UNK_TYPE D_0C006C00;
+typedef enum {
+    /* 0 */ EQUIP_STATE_MAGIC_ARROW_GROW_ORB,
+    /* 1 */ EQUIP_STATE_MAGIC_ARROW_MOVE_TO_BOW_SLOT,
+    /* 2 */ EQUIP_STATE_MAGIC_ARROW_HOVER_OVER_BOW_SLOT,
+    /* 3 */ EQUIP_STATE_MOVE_TO_C_BTN
+} EquipState;
 
-s32 D_8082B3C0 = 0;
+s16 sEquipState = EQUIP_STATE_MAGIC_ARROW_GROW_ORB;
 
-s32 D_8082B3C4 = 0;
+// Timer to hold magic arrow icon over magic arrow slot before moving when equipping.
+s16 sEquipMagicArrowSlotHoldTimer = 0;
 
-s16 D_8082B3C8 = 10;
+// Number of frames to move icon from slot to target position when equipping.
+s16 sEquipAnimTimer = 10;
 
-u8 D_8082B3CC[5][24] = {
+u8 gPlayerFormSlotRestrictions[PLAYER_FORM_MAX][24] = {
```

Comment:
I suppose there's a question of whether 24 should have a define, but it's probably mega-hardcoded.

Reply:
I've thought about it, but the 24 applies to both the 24 items and the 24 masks, and you'll see this constant pop up a lot in the next "mask page" pr as well, so it'd have to be general. I'm undecided on it, but it would have to capture both items and masks in some way

Reply:
Yeah, and it could just as well be `6 * 4` too.

Reply:
I might make that part of the following masks PR then if that's okay? It applies well to that PR and it would be better to see the define in the full context of both items and masks

Reply:
Yeah, just wanted to point it out before I forget about it, and so that you have it in mind for subsequent work.